src/libasr/asr_utils.h

```
@@ -112,6 +112,15 @@ static inline ASR::symbol_t *symbol_get_past_external(ASR::symbol_t *f)
     }
 }
 
+template <typename T>
+Location get_vec_loc(const Vec<T>& args) {
```

I'm not really a fan of this, since these kind of changes to `loc` can easily introduce bugs in code.

I'll rethink to introduce to have `loc` associated with `Vec` itself.

---

What kind of bugs can this introduce? It seems this is ok.

---

I don't like the idea of changing "loc" at some random point in the code, like it's being done here. I think a good place for assigning "loc" would be when parsing (i.e. in `semantics.h`), so that "loc" is assigned to "Vector" object itself and bound to it, instead of separately writing a function for that.

Let's say if I want to raise an error for "vector of arguments" again somewhere else, I would call this function again, that's something I don't really like. Having "loc" bound to "Vector of arguments" is too obvious for anyone new to the code as well.

---

I see. I think it's fine. The downside of always computing location every time you create a vector is that it will slow things down. We don't report error messages that often.